hydromt/components/mesh.py

```diff
@@ -680,6 +719,50 @@ def _grid_is_equal(self, grid_name: str, data: xu.UgridDataset) -> bool:
             .equals(data.grid.to_dataset(optional_attributes=True))
         )
 
+    def _get_mesh_grid_data(self, grid_name: str) -> Union[xu.Ugrid1d, xu.Ugrid2d]:
+        if self._region_component is not None:
+            region_component = cast(
+                MeshComponent, self.model.get_component(self._region_component)
+            )
+            if region_component is None:
+                raise ValueError(
+                    f"Unable to find the referenced region component: '{self._region_component}'"
+                )
```

Comment:
Same as in Grid, I think it's good to check if the component is actually a mesh and raise if it's not

Reply:
Same reply as above. What does cast actually do if not type checking already?